cli/commands/terraform/download_source.go

```diff
@@ -102,7 +103,9 @@ func downloadTerraformSourceIfNecessary(terraformSource *terraform.Source, terra
 	})
 
 	if downloadErr != nil {
-		return downloadErr
+		download_info := fmt.Errorf("Downloading Terraform source from %s", terraformSource.CanonicalSourceURL.String())
```

Comment:
* why can't be created specific error struct like `WorkingDirNotFound`?
* use camel case instead of snake case 
* why can't be used `go-multierror` to combine multiple errors?

Reply:
Linter suggestion "Error string should not be capitalized or end with punctuation mark "

Reply:
thanks, will update it accordingly.

Reply:
if we use go-multierror, we got something as below:
![image](https://github.com/gruntwork-io/terragrunt/assets/34583489/f691749d-6e76-4184-b7a8-ad7c721d4ee4)
Actually, there is only 1 error.
If we keep the current code, we get something as below:
<img width="1088" alt="image" src="https://github.com/gruntwork-io/terragrunt/assets/34583489/98c0e3f7-c23c-4555-ac63-e64478ba0c23">

Reply:
For creating a struct, since we only need the 'Url' string, in my opinion, we do not need a struct in this case. just my opinion. please review and advise. thanks,

Reply:
@denis256 updated as you suggested. added a struct 'DownloadingTerraformSourceNotFound', removed the snake case, and updated the test accordingly. please review and advise. thanks.

Reply:
I think error name `DownloadingTerraformSourceNotFound` is too specific ( "NotFound" part) - downloadErr can be returned because of 404/500 etc...

Reply:
updated error name 'DownloadingTerraformSourceNotFound'  to 'DownloadingTerraformSourceErr'.